### ISSUE TITLE
Array broadcasting issues

### ISSUE THREAD

**Issue by DamienIrving** [SciTools/iris#2765]
I recently upgraded to iris version 1.13.0 and simple array broadcasting operations that worked with previous versions aren't working now. For instance, I'm getting the following error:
```
Traceback (most recent call last):
  File "/home/599/dbi599/ocean-analysis/data_processing/calc_ohc.py", line 217, in <module>
    main(args)
  File "/home/599/dbi599/ocean-analysis/data_processing/calc_ohc.py", line 153, in main
    temperature_cube = temperature_cube * area_cube
  File "/g/data/r87/dbi599/miniconda3/envs/ocean/lib/python3.6/site-packages/iris/analysis/maths.py", line 336, in multiply
    in bad_coord_grps}))
ValueError: This operation cannot be performed as there are differing coordinates (time, depth) remaining which cannot be ignored.
```
The temperature cube looks like this:
```
print(temperature_cube)
sea_water_potential_temperature / (K) (time: 120; depth: 31; latitude: 189; longitude: 192)
     Dimension coordinates:
          time                             x           -             -               -
          depth                            -           x             -               -
          latitude                         -           -             x               -
          longitude                        -           -             -               x
```

and the area cube like this:
```
print(area_cube)
cell_area / (m2)                    (latitude: 189; longitude: 192)
     Dimension coordinates:
          latitude                           x               -
          longitude                          -               x

```

**Comment by ajdawson** [comment]
Related: #1887 #2599.

**Comment by rcomer** [comment]
This should be sorted for Iris v3.  See #3774 and #3785.

**Comment by bjlittle** [comment]
Closed by #3785
